### PR TITLE
Rocky/add category field

### PR DESCRIPTION
**Pull request by rocky-fleek**
## Summary
Added `category` field in canister metadata. 
The `category` is one of 3 possible values (Asset, Nft, Service) and so I implemented it as an enum but I'm not sure if that's correct or not. 

@Nima-Ra I know I'm missing something on the upgrade script, cause I don't know how it would populate the fields otherwise.  Just not sure how you do that to be honest

### CODE REVIEW

**Comment by rocky-fleek** [comment]
I just noted that the csv already has a category field.  I'll add other types and change the ones I used tomorrow :catsalute:

File: candid/registry.did [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+    Service;
+};
+
 type canister_metadata = record {
     name: text;
     description: text;
     url: text;
     logo_url: text;
     version: nat32;
+    category: canister_category
```
Comment (Nima-Ra) [inline candid/registry.did:13]: missing semicolon

File: candid/registry.did [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 };
 
 type input_canister_metadata = record {
     name: text;
     description: text;
     url: text;
     logo_url: text;
+    category: canister_category
```
Comment (Nima-Ra) [inline candid/registry.did:21]: missing semicolon

**Review by Nima-Ra — COMMENTED** [review]
Looks good to me! I see you have come far in learning rust!
I don't think we should merge until the upgrade.rs is updated.
I will update that later and merge ;)

File: registries/canister_registry/src/upgrade.rs [hunk 3]
```diff
@@ -30,8 +150,21 @@ pub fn pre_upgrade() {
 
 #[post_upgrade]
 pub fn post_upgrade() {
-    if let Ok((stable,)) = ic::stable_restore::<(StableStorage,)>() {
-        ic::get_mut::<CanisterDB>().load(stable.db);
+    if let Ok((stable,)) = ic::stable_restore::<(StableStorageV0,)>() {
+        let data = get_data();
+        let mut canister_list = Vec::with_capacity(stable.db.len());
+        for (_key, canister_info) in stable.db.into_iter().enumerate() {
```
Comment (qti3e) [inline registries/canister_registry/src/upgrade.rs:156]: Unnecessary `enumerate` if you don't need the `_key`

**Pull request merged**